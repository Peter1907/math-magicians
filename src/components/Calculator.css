.container {
  text-align: center;
  display: flex;
  flex-direction: column;
  width: 500px;
  margin: auto;
}

.display {
  background-color: #afafaf;
  height: 70px;
  font-size: 2em;
  padding: 10px 40px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
}

.keys {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
}

.key {
  margin: 0;
  padding: 30px;
  background-color: #e9e9e9;
  font-size: 18px;
}

.zero {
  grid-column: 1 / 3;
}

.symbol {
  background-color: orange;
}
