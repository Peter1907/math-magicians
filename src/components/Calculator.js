import React, { useState } from 'react';
import Key from './Key';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const { total, operation, next } = state;
  return (
    <div className="container">
      <h2 style={{ fontSize: '32px' }}>Let&apos;s do some math!</h2>
      <div className="calculator">
        <div className="display">
          <p>{total || ''}</p>
          <p>{operation || ''}</p>
          <p>{next || ''}</p>
        </div>
        <div className="keys">
          <Key classes="key" name="AC" state={state} setState={setState} />
          <Key classes="key" name="+/-" state={state} setState={setState} />
          <Key classes="key" name="%" state={state} setState={setState} />
          <Key classes="key symbol" name="รท" state={state} setState={setState} />
          <Key classes="key" name="7" state={state} setState={setState} />
          <Key classes="key" name="8" state={state} setState={setState} />
          <Key classes="key" name="9" state={state} setState={setState} />
          <Key classes="key symbol" name="&times;" state={state} setState={setState} />
          <Key classes="key" name="4" state={state} setState={setState} />
          <Key classes="key" name="5" state={state} setState={setState} />
          <Key classes="key" name="6" state={state} setState={setState} />
          <Key classes="key symbol" name="-" state={state} setState={setState} />
          <Key classes="key" name="1" state={state} setState={setState} />
          <Key classes="key" name="2" state={state} setState={setState} />
          <Key classes="key" name="3" state={state} setState={setState} />
          <Key classes="key symbol" name="+" state={state} setState={setState} />
          <Key classes="key zero" name="0" state={state} setState={setState} />
          <Key classes="key" name="." state={state} setState={setState} />
          <Key classes="key symbol" name="=" state={state} setState={setState} />
        </div>
      </div>
    </div>
  );
};

export default Calculator;
