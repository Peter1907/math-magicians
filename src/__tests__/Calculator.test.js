/* eslint-disable*/

import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

// This file is testing both of the Calculator and the Key components,
// because the Calculator is depending on the Key component to work properly.
it('renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});