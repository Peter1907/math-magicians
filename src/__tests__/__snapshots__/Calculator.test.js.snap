// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <h2
    style={
      Object {
        "fontSize": "32px",
      }
    }
  >
    Let's do some math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
      <p />
      <p />
      <p />
    </div>
    <div
      className="keys"
    >
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="key symbol"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="key symbol"
        onClick={[Function]}
        type="button"
      >
        ร
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="key symbol"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="key symbol"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="key zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="key symbol"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
